### PR TITLE
Relocate dependencies

### LINKED ISSUE #539: IllegalStateException on Gradle Task: deobfMcSRG
I've run setupDecompWorkspace and I get hit with this:

https://pastebin.com/RUJTUzjU

Here is my build.gradle:

https://pastebin.com/5PwbuyLA

Iv'e tried just about everything. --refresh-dependencies, cleared the cache, deleted the cache folder, deleted the .gradle and nothing seems to fix this. My imports have also broken. Im not sure what else to do at this point.

### PR DESCRIPTION
**Pull request by 14mRh4X0r**
This pull request aims to resolve #539.

### CODE REVIEW

**Comment by 14mRh4X0r** [comment]
Updating Gradle is necessary for the newest Shadow plugin to work. The newest version compatible with Gradle 2.14, Shadow 1.2.4, clashes with the ASM in ForgeGradle itself, as noted in #542.

**Comment by liach** [comment]
do we really want to shadow/ship gradle api?

**Comment by kashike** [comment]
No.

**Comment by 14mRh4X0r** [comment]
The configuration `shadow` is a somewhat misleading name; it means those won't be shaded into the jar. See https://imperceptiblethoughts.com/shadow/configuration/#configuring-the-runtime-classpath.

You'll also notice I replaced all `compile` with `shadow` (not included in the jar), and all `shade` with `implementation` (former `compile`, shaded into the jar).

**Comment by DaemonUmbra** [comment]
Consider using [Forge’s Jar-in-Jar](https://mcforge.readthedocs.io/en/latest/gettingstarted/dependencymanagement/#dependency-extraction) system rather than shadow.

**Comment by 14mRh4X0r** [comment]
That's not an option when you're not actually using Forge, e.g. when using [VanillaGradle](//github.com/Minecrell/VanillaGradle)

**Comment by liach** [comment]
jar in jar does not apply to gradle plugins as there's no need to disable one of the jars shipped.

**Comment by liach** [comment]
for the configuration name, blame shadow!

**Comment by LexManos** [comment]
The real issue is that we're packaging ASM at all.
IIRC, the packaging was meant for things that are not on maven central {SpecialSource of most note, and Artifactual for FG3+}
Everything else should be possible to standard dependencies that gradle manages.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Sat Aug 20 15:56:50 PDT 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-bin.zip
```
Comment (LexManos) [inline gradle/wrapper/gradle-wrapper.properties:5]: No. Don't try sneaking this crap in. This breaks peoples setups.

Comment (14mRh4X0r) [reply gradle/wrapper/gradle-wrapper.properties:5]: I'll quote my comment:

> Updating Gradle is necessary for the newest Shadow plugin to work. The newest version compatible with Gradle 2.14, Shadow 1.2.4, clashes with the ASM in ForgeGradle itself, as noted in #542.

Comment (LexManos) [reply gradle/wrapper/gradle-wrapper.properties:5]: Yes, I saw that, it shouldn't be an issue tho. With the proper non-packaging of ASM.

**Comment by AbrarSyed** [comment]
IIRC I started doing this because at some point an update in special source broke everything, so I did this as a way to pin the version to that specific FG deployment. At the time we were using SS snapshots

On January 7, 2019 7:20:00 PM UTC, LexManos <notifications@github.com> wrote:
>The real issue is that we're packaging ASM at all.
>IIRC, the packaging was meant for things that are not on maven central
>{SpecialSource of most note, and Artifactual for FG3+}
>Everything else should be possible to standard dependencies that gradle
>manages.
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/MinecraftForge/ForgeGradle/pull/542#issuecomment-452049668

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by DaemonUmbra** [comment]
Closing this PR as it does not target the currently supported ForgeGradle branch (``FG_3.0``).

Note that an ongoing effort to update legacy MinecraftForge branches to utilize FG_3.0 is ongoing and accepting informed contributions. (For more information [join the Discord](https://discord.gg/UvedJ9m) and  read through the #mcpconfig channel)

**Pull request closed without merging**